optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -129,7 +129,12 @@ void recordAndRegisterBeans(OptaPlannerRecorder recorder, RecorderContext record
         if (indexView.getAnnotations(DotNames.PLANNING_SOLUTION).isEmpty()
                 && indexView.getAnnotations(DotNames.PLANNING_ENTITY).isEmpty()) {
             log.warn("Skipping OptaPlanner extension because there are no " + PlanningSolution.class.getSimpleName()
-                    + " or " + PlanningEntity.class.getSimpleName() + " annotated classes.");
+                    + " or " + PlanningEntity.class.getSimpleName() + " annotated classes."
+                    + "\nIf your domain classes are located in a dependency of this project, maybe try generating"
+                    + " the Jandex index by adding the following entries to the application.properties:"
+                    + "\nquarkus.index-dependency.<name>.group-id="
+                    + "\nquarkus.index-dependency.<name>.artifact-id="
+                    + "\nSee https://quarkus.io/guides/cdi-reference#how-to-generate-a-jandex-index for more details.");
```

Comment:
Not sure if we want to put a quarkus guides URL in our error message - they might end up giving 404's without us ever noticing it.

Normally we keep the "maybe" line on a single line, in case there are multiple "maybe" lines. Not sure if that would be a good idea here: `... by adding application.properties entries (quarkus.index-dependency.<name>.group-id and quarkus.index-dependency.<name>.artifact-id).`

Reply:
ad guide URL: the extra lines in application.properties might not be always the best solution. If a user controls the dependency with domain classes, adding the maven plugin to that maven project/module is a better option as it doesn't require leaking the GAV from pom.xml to application.properties.

ad formatting: I see, both have their own pros and cons.

Reply:
Changed the suggestion to include the maven plugin. URL has been removed.